pkg/agent/run.go

```diff
@@ -437,55 +437,22 @@ func postData(ctx context.Context, config CombinedConfig, preflightClient client
 		log.Info("Data sent successfully")
 
 		return nil
-	}
-
-	if config.OrganizationID == "" {
```

Comment:
This will not be true when `TLSPKMode` is `JetstackSecureOAuth` or `JetstackSecureAPIToken`, since this is checked here: https://github.com/jetstack/jetstack-secure/blob/5aab7339bbd84cdcb623db2f8fcea5969d65363a/pkg/agent/config.go#L598-L600